StripeCore/StripeCore/Source/Attestation/StripeAttest.swift

```diff
@@ -18,18 +18,31 @@ import UIKit
 
     /// Sign an assertion.
     /// Will create and attest a new device key if needed.
-    @_spi(STP) public func assert() async throws -> Assertion {
+    /// Returns an AssertionHandle, which must be called after the network request completes (with success or failure) in order to unblock future assertions.
+    @_spi(STP) public func assert() async throws -> AssertionHandle {
+        // Make sure we only process one assertion at a time, until the latest
+        if assertionInProgress {
+            try await withCheckedThrowingContinuation { continuation in
+                assertionWaiters.append(continuation)
+            }
+        }
+        assertionInProgress = true
+
         do {
             let assertion = try await _assert()
             let successAnalytic = GenericAnalytic(event: .assertionSucceeded, params: [:])
-            STPAnalyticsClient.sharedClient.log(analytic: successAnalytic, apiClient: apiClient)
-            return assertion
+            if let apiClient {
+                STPAnalyticsClient.sharedClient.log(analytic: successAnalytic, apiClient: apiClient)
+            }
+            return AssertionHandle(assertion: assertion, stripeAttest: self)
         } catch {
             let errorAnalytic = ErrorAnalytic(event: .assertionFailed, error: error)
```

Comment:
If there is an exception, are users expected to call `complete()`? If so, that seems odd.  Seems like if there is an exception, we should set assertionInProgress to false and continue with the next one that is queue'd up?

Reply:
Good catch! Will fix this and add a test.